### DISCUSSION TITLE
Smart Sync type support?

### DISCUSSION THREAD

**Discussion by jeffkruczynski** [samschott/maestral#444] [Q&A]
Hello,
I heard about this project on the Mac Observer podcast [Mac Geek Gab](https://www.macobserver.com/show/mac-geek-gab/), and was intrigued as I had been noticing a good amount of memory usage by the Dropbox client on my M1 MacBook Air.
This project looks awesome! I do have a couple quick questions that I haven't been able to answer so far looking through the app documentation and the project site, so I was hoping someone here could verify:
- I don't see that a Smart Sync type functionality exists in Maestral right now; is that correct? (my use case for this is saving SSD space on my devices while still seeing and having access to the content that's in my Dropbox account)
- Is there a recommended transition path for moving a device from the Dropbox app to Maestral? I saw the warning not to run both simultaneously; but, for example, is it "safe" to simply uninstall the Dropbox client, then install Maestral and point it to the folder that the Dropbox app had been using? Or is it preferable to uninstall Dropbox, delete the folder, and start from scratch with Maestral? (Especially given that I have Smart Sync enabled in the Dropbox app, so many of the files in the Dropbox folder are not actually present on the device...I don't want that to confuse Maestral and end up with a bunch of deleted files)
Thanks!
Jeff

**Comment by samschott** [comment]
Hi Jeff,

I did not realise that this project has been discussed in Mac podcasts :) It really was targeted towards Linuxy command line users, but seems to have found some traction on macOS as well.

To answer your questions:

> I don't see that a Smart Sync type functionality exists in Maestral right now; is that correct?

Correct. Smart sync requires a level of platform and file system integration which I have not looked into yet. If this should ever become a feature of Maestral, it will likely be limited to specific file systems, such as Apple's APFS.

> is it "safe" to simply uninstall the Dropbox client, then install Maestral and point it to the folder that the Dropbox app had been using?

This is safe in principle. You can select the existing folder in the setup dialog and choose "Merge". This will reindex its content but won't download any existing files, therefore saving both a lot of time and bandwidth.

However, with smart sync turned on, local files will only be "stubs" without content and will therefore trigger sync conflicts and resulting "conflicting copies" when Maestral realises that their file content differs from what it sees on the server.

If you do want to transition, I'd recommend downloading all files which you want to keep locally and deleting / excluding all other files. If you did rely on Smart Sync and want to continue using it, I'm afraid you'll need to stick to the official client.

**Comment by jeffkruczynski** [comment]
Hi Sam - thank you for the followup! That’s what I figured. I’ll have to decide whether selective sync or smart sync is preferable. In any event, it’s an awesome tool so thank you for all the work you do on it.
Jeff
